# ferrari/config/controllers_op.yaml

controller_manager:
  ros__parameters:
    use_sim_time: true
    update_rate: 100

    joint_state_broadcaster:
      type: joint_state_broadcaster/JointStateBroadcaster

    f1tenth_commands_controller:
      type: F1TenthCommandsController

/ferrari_op/controller_manager:
  ros__parameters:
    use_sim_time: true
    update_rate: 100

    joint_state_broadcaster:
      type: joint_state_broadcaster/JointStateBroadcaster

    f1tenth_commands_controller:
      type: F1TenthCommandsController

f1tenth_commands_controller:
  ros__parameters:
    left_steer_joint:  "front_left_steer_joint"
    right_steer_joint: "front_right_steer_joint"
    left_wheel_joint:  "rear_left_wheel_joint"
    right_wheel_joint: "rear_right_wheel_joint"

    servo_center: 0.5
    steer_max_left_rad:  -0.5236
    steer_max_right_rad: -0.5236
    servo_axis_inverted: false
    erpm_per_radps: 300.0

/ferrari_op/f1tenth_commands_controller:
  ros__parameters:
    left_steer_joint:  "front_left_steer_joint"
    right_steer_joint: "front_right_steer_joint"
    left_wheel_joint:  "rear_left_wheel_joint"
    right_wheel_joint: "rear_right_wheel_joint"

    servo_center: 0.5
    steer_max_left_rad:  -0.5236
    steer_max_right_rad: -0.5236
    servo_axis_inverted: false
    erpm_per_radps: 300.0
